lib/tokenizers/added_token.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Tokenizers.AddedToken do
+  @moduledoc """
+  This struct represents AddedTokens
+  """
+
+  @type t() :: %__MODULE__{resource: reference()}
+  defstruct [:resource]
+
+  @doc """
+  Create a new AddedToken. You can choos if it's special or not.
+
+  It's created with defaults:
+
+  * `single_word` is `false`
+  * `lstrip` is `false`
+  * `rstrip` is `false`
+  * `normalized` is `true` for not special tokens, `false` for special tokens.
+  """
+  @spec new(token :: String.t(), is_special :: boolean()) :: t()
+  defdelegate new(token, is_special \\ false), to: Tokenizers.Native, as: :added_token_new
```

Comment:
What if we make our API be:

    Tokenizers.AddedToken.new(token, single_word: true | false, lstrip: true | false, rstrip: true | false, ...)

It is a smaller API surface and it exposes less mutability.

Reply:
We can make immutable a lot, the problem is that I don't have enough expertise in DS - is this part in general should me mutable, or it's just the implementation details in Rust?
I mean - if it's possible to change the token AFTER it's applied to trainer? If not - then we can make things really simple. WDYT?

Reply:
I don't see a reason to mutate it after it is applied to the trainer. I would prefer to not offer any mutability API at all. We create it once with all of the desired attributes set and that's it. :)